firebase-ai/src/main/kotlin/com/google/firebase/ai/FirebaseAI.kt

```diff
@@ -123,21 +134,23 @@ internal constructor(
         """Unsupported Gemini model "$modelName"; see
       https://firebase.google.com/docs/vertex-ai/models for a list supported Gemini model names.
       """
-          .trimIndent()
+          .trimIndent(),
       )
     }
-    if (location.trim().isEmpty() || location.contains("/")) {
-      throw InvalidLocationException(location)
-    }
     return LiveGenerativeModel(
-      "projects/${firebaseApp.options.projectId}/locations/${location}/publishers/google/models/${modelName}",
+      when (backend.backend) {
+        GenerativeBackendEnum.VERTEX_AI ->
+          "projects/${firebaseApp.options.projectId}/locations/${backend.location}/publishers/google/models/${modelName}"
+        GenerativeBackendEnum.GOOGLE_AI ->
+          throw ServiceDisabledException("Live Model is not yet available on the Google AI backend")
```

Comment:
Not 100% sold that `ServiceDisabledException` since it points to https://firebase.google.com/docs/vertex-ai/faq-and-troubleshooting?platform=android#required-apis as a way to address them